### ISSUE TITLE
When using server-side pairing, the SIN is different than expected when uncompressed public key is used

### ISSUE THREAD

**Issue by CherryDT** [btcpayserver/btcpayserver#3432]
The [docs](https://docs.btcpayserver.org/CustomIntegration/#server-side-pairing) say:

> The second way, is to generate your private key via some bitcoin library then:
> 
> 1. Go to the store's settings
> 2. Click on Access tokens
> 3. Click on Create new Token
> 4. Select merchant's facade and enter your public key,
> 5. Click request pairing
> 6. Click on Approve

Note that they don't specify whether the provided pubkey should be compressed or uncompressed. If we take the [BitPay setup script](https://github.com/bitpay/nodejs-bitpay-client/blob/master/src/setup/BitPaySetup.ts) for example, it will output the public key with `keyPair.getPublic('hex')` which results in an uncompressed key. Of course the same works with the `btcpay` SDK, so what I did was:

```javascript
const btcpay = require('btcpay')
const keyPair = btcpay.crypto.generate_key_pair()
console.log('Public key', keyPair.getPublic('hex'))
console.log('SIN', btcpay.crypto.get_sin_from_key(keyPair))
```

...which resulted in:

```
0459cb199455a288c6e6b032da21d902330fc8d38eccfab25495905dfb0a61e82e7eddb3897af2ad3ebfa651608b9f264d76fc562c99f85a01c63f1922f01247ca
TfGgz8qVP39nrZLQs7VKJ3AJLq32QKV3bHa
```

However, to my surprise, putting this into the BTCPay "Add token" page, I got a different SIN displayed:

![image](https://user-images.githubusercontent.com/6010463/152860171-c0d9a8f8-00ee-44a4-ba63-c4141adee5c7.png)
![image](https://user-images.githubusercontent.com/6010463/152860195-99576a1c-b18e-4d41-be1a-4877b1219f58.png)

I wasn't sure where this other SIN `TfGyEvDV848jjiycGgfMy3ipAG7jv1D3vwU` came from.

Turns out the issue is that I put in an **uncompressed** public key. Putting in the **compressed** version of the public key (using `keyPair.getPublic().encodeCompressed('hex')` instead of `keyPair.getPublic('hex')`), `0259cb199455a288c6e6b032da21d902330fc8d38eccfab25495905dfb0a61e82e`, results in the correct SIN to be displayed.

Now, all of the client libraries I looked at will ensure that the compressed private key is used in the SIN algorithm. For example the `btcpay` node.js libary does it [here](https://github.com/btcpayserver/node-btcpay/blob/92cfe4342c018e4329b4653e426a98a6a102697c/src/core/cryptography.ts#L19-L26
) with `encodeCompressed`.

I believe that it shouldn't matter whether the compressed or the uncompressed public key is entered into the UI, the SIN should be the same (based on the compressed key like the client libraries also calculate it) either way.

**Comment by NicolasDorier** [comment]
@CherryDT can you tell us why you had a uncompressed pubkey in the first place?

**Comment by NicolasDorier** [comment]
oh ok this is the btcpay lib, we need to fix this asap.

**Comment by NicolasDorier** [comment]
@CherryDT  This package is deprecated, BTCPay Server is exposing a new, more complete and easy to use API called Greenfield. [Check the doc](https://docs.btcpayserver.org/Development/GreenFieldExample/)

**Comment by CherryDT** [comment]
Hi, I know there is the Greenfield API. However it is lacking functionality compared to the BitPay API (in particular per-transaction webhook URLs), and it seems compatibility with BitPay is still advertised for BTCPay.

Anyway, since given only a private key (or key pair object) any client library I looked at (JS, PHP, Go) uses a compressed key automatically, and the docs of most also specify that only a compressed key should be passed to the SIN-generation function in case it takes a single public key and not a key pair, it seems that anybody submitting an uncompressed key here in the UI is doing so in error, as I did.

It shouldn't matter because the docs say that "any bitcoin library" can be used, and any library I know allows to print either the compressed or uncompressed key depending on how you call it, so the assumption can't be made that the key is always compressed when provided by the user for server-side pairing and not via client-side pairing.
